Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -1342,7 +1342,7 @@ BTDiagnostics::InterleaveBufferAndSnapshotHeader ( const std::string& buffer_Hea
     amrex::IntVect box_lo(0);
     amrex::IntVect box_hi(1);
     // Update prob_lo with min of buffer and snapshot
-    for (int idim = 0; idim < snapshot_HeaderImpl.spaceDim(); ++idim) {
+    for (int idim = 0; idim < std::min(snapshot_HeaderImpl.spaceDim(),AMREX_SPACEDIM); ++idim) {
```

Comment:
Good solution, but a bit cryptic :)

Could be a separate variable before the loop with an inline comment why we do this...

Or we just extend the compiler flags...